src/konnektor/network_planners/_map_scoring.py

```diff
@@ -54,8 +54,8 @@ def _determine_best_mapping(
 
             if len(tmp_mappings) > 0:
                 tmp_best_mapping = max(tmp_mappings, key=lambda m: m.annotations["score"])
-
-                if tmp_best_mapping.annotations["score"] > best_score:
+                # TODO: we still need a tie-breaking scheme, then we can get rid of 'best_mapping is None'
+                if tmp_best_mapping.annotations["score"] > best_score or best_mapping is None:
```

Comment:
I changed this in https://github.com/OpenFreeEnergy/konnektor/pull/138 and now realize that was a mistake...

Reply:
🤔 If a score is 0, do we assume this to be a non-feasible mapping or is that allowable?

It seems to me like this `best_mapping is None` behaviour only exists for that one edge case, where I would intuitively expect a 0 score to be "this isn't a mapping, ignore it".

Reply:
we say everywhere that a scorer must return between 0 and 1, where 0 is the worst and 1 is the best, so I think 0 should be treated as valid.

Reply:
Yeah here we go; https://github.com/OpenFreeEnergy/Lomap/blob/main/lomap/gufe_bindings/network_generation.py#L95

Looks like it suffers from the same missing bit, we should update it to match behaviour.

Reply:
I think it's more rigorous to allow 0.0 and 1.0 as the extremes of "good" and "bad" - i'm leaning toward:

> starting condition of best_score = -1.0 instead of 0.0, so that any valid score (between 0 and 1, something we should validate) will beat out the initial condition.

do you agree?

Reply:
in an offline conversation, @IAlibay and I decided that we should:
  1. keep the `best_mapping is None` check here, since we don't (yet) enforce a score being in [0,1]
  2. as a follow-up, add the [0,1] enforcement here, and update tests accordingly (both here and in openfe, where we have scores >1 in our test suite).

Reply:
done, and opened an issue for 2 here: https://github.com/OpenFreeEnergy/konnektor/issues/143